lib/re1.5/compilecode.c

```diff
@@ -18,19 +18,21 @@ static void _emit_checked(int at, char *code, int val, bool *err) {
     }
 }
 
-static const char *_compilecode(const char *re, ByteProg *prog, int sizecode)
+static const char *_compilecode(const char *re, size_t *len, ByteProg *prog, int sizecode)
 {
     char *code = sizecode ? NULL : prog->insts;
     bool err = false;
     int start = PC;
     int term = PC;
     int alt_label = 0;
 
-    for (; *re && *re != ')'; re++) {
+    
+    while (*len > 0 && *re != ')') {
```

Comment:
As above, might be easier to have `re_top` variable instead of `len`, and do `re < re_top` (maybe...).